static/app/views/nav/secondary/sections/issues/issueViews/issueViewNavEllipsisMenu.tsx

```diff
@@ -1,74 +1,80 @@
+import {useLocation} from 'sentry/utils/useLocation';
 import {useNavigate} from 'sentry/utils/useNavigate';
 import useOrganization from 'sentry/utils/useOrganization';
-import type {IssueView} from 'sentry/views/issueList/issueViews/issueViews';
+import {useParams} from 'sentry/utils/useParams';
+import {createIssueViewFromUrl} from 'sentry/views/issueList/issueViews/createIssueViewFromUrl';
+import {useIssueViewUnsavedChanges} from 'sentry/views/issueList/issueViews/useIssueViewUnsavedChanges';
+import {useCreateGroupSearchView} from 'sentry/views/issueList/mutations/useCreateGroupSearchView';
+import {useDeleteGroupSearchView} from 'sentry/views/issueList/mutations/useDeleteGroupSearchView';
+import {useUpdateGroupSearchView} from 'sentry/views/issueList/mutations/useUpdateGroupSearchView';
+import type {NavIssueView} from 'sentry/views/nav/secondary/sections/issues/issueViews/issueViewNavItems';
 
 export interface IssueViewNavEllipsisMenuProps {
-  baseUrl: string;
+  // TODO(msun): Allow deleting last view once horizontal tab views are deleted
   isLastView: boolean;
-  onDeleteView: () => void;
-  onDuplicateView: () => void;
-  onUpdateView: (view: IssueView) => void;
   setIsEditing: (isEditing: boolean) => void;
-  view: IssueView;
+  view: NavIssueView;
   sectionRef?: React.RefObject<HTMLDivElement | null>;
 }
 
 export function IssueViewNavEllipsisMenu({
   sectionRef,
   setIsEditing,
   view,
-  onDeleteView,
-  onDuplicateView,
-  onUpdateView,
-  baseUrl,
   isLastView,
 }: IssueViewNavEllipsisMenuProps) {
   const navigate = useNavigate();
   const organization = useOrganization();
+  const location = useLocation();
+  const {viewId} = useParams<{viewId: string}>();
+  const isSelected = viewId === view.id;
 
-  const handleSaveChanges = () => {
-    const updatedView = {
-      ...view,
-      query: view.unsavedChanges?.query ?? view.query,
-      querySort: view.unsavedChanges?.querySort ?? view.querySort,
-      projects: view.unsavedChanges?.projects ?? view.projects,
-      environments: view.unsavedChanges?.environments ?? view.environments,
-      timeFilters: view.unsavedChanges?.timeFilters ?? view.timeFilters,
-      unsavedChanges: undefined,
-    };
-    onUpdateView(updatedView);
-    navigate(constructViewLink(baseUrl, updatedView));
+  const {mutate: updateIssueView} = useUpdateGroupSearchView({
+    onSuccess: () => {
+      trackAnalytics('issue_views.saved_changes', {
+        leftNav: true,
+        organization: organization.slug,
+      });
+    },
+  });
 
-    trackAnalytics('issue_views.saved_changes', {
-      leftNav: true,
-      organization: organization.slug,
-    });
-  };
+  const {mutate: deleteIssueView} = useDeleteGroupSearchView({
+    onSuccess: () => {
+      trackAnalytics('issue_views.deleted_view', {
+        leftNav: true,
+        organization: organization.slug,
+      });
+    },
+    onError: () => {
+      addErrorMessage(t('Failed to delete view'));
+    },
+  });
```

Comment:
i think i would not pass these in as options and use mutate's 2nd arguments to handle these

deleteIssueView(args, {onSuccess: () => {})